stable_plugins/classical_ml/data_preparation/pandas_preprocessing/tasks.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2023 QHAna plugin runner contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from tempfile import SpooledTemporaryFile
+
+from typing import Optional
+
+from celery.utils.log import get_task_logger
+
+from . import PDPreprocessing
+
+from .schemas import (
+    FirstInputParameters,
+    FirstInputParametersSchema,
+    SecondInputParameters,
+    SecondInputParametersSchema,
+)
+from qhana_plugin_runner.celery import CELERY
+from qhana_plugin_runner.db.models.tasks import ProcessingTask
+from qhana_plugin_runner.storage import STORE
+
+from pandas import read_csv
+from pretty_html_table import build_table
+from .backend.checkbox_list import get_checkbox_list_dict
+from .backend.optionlist import get_optionlist
+
+TASK_LOGGER = get_task_logger(__name__)
+
+
+@CELERY.task(name=f"{PDPreprocessing.instance.identifier}.first_task", bind=True)
+def first_task(self, db_id: int) -> str:
+    # get parameters
+
+    TASK_LOGGER.info(
+        f"Starting new pandas preprocessing calculation task with db id '{db_id}'"
+    )
+    task_data: Optional[ProcessingTask] = ProcessingTask.get_by_id(id_=db_id)
+
+    if task_data is None:
+        msg = f"Could not load task data with id {db_id} to read parameters!"
+        TASK_LOGGER.error(msg)
+        raise KeyError(msg)
+
+    input_params: FirstInputParameters = FirstInputParametersSchema().loads(
+        task_data.parameters
+    )
+
+    file_url = input_params.file_url
+
+    TASK_LOGGER.info(f"Loaded input parameters from db: {str(input_params)}")
+
+    df = read_csv(file_url)
+
+    # Output data
+    with SpooledTemporaryFile(mode="w") as output:
+        df.to_csv(output, index=False)
+        task_file = STORE.persist_task_result(
+            db_id,
+            output,
+            "file.csv",
+            "entity",
+            "text/csv",
+        )
+        task_data.data["file_url"] = task_file.file_storage_data
+        # task_data.data["pandas_html"] = df.to_html(max_rows=100)
+        task_data.data["pandas_html"] = build_table(df, "grey_light")
```

Comment:
This might get larger than what a single row in the Database can store. Maybe use the blob storage table instead.

Reply:
This is a rendered html of the retrieved table. It will be displayed on the MicroFrontend, to help the user decide which preprocessing step he wants to take next. If this is too big to save "normaly" within the database, it will probably also be too big, to render on the MicroFrontend and still be of use to a user, right?
Maybe instead of using the blob format, the displayed amount of the table should be limited and thus, avoiding a big rendered html code. What do you think?

Reply:
That could also work. But make sure to display an error message to the user in case the preview was too large.

Reply:
I can limit the number of rows and columns that will be included in the rendered html of the table, to avoid large storage usage. This doesn't garuantee that the rendered html will be small, though, e.g. if the dataframe has only one row and one column, with a realy big entry. Do you have any suggestions on how to deal with this?

Reply:
You could check the html length before comitting to the DB and if the HTML is too large commit a fallback html instead. Or be creative with shortening values. But generally do not put too much work into that safety feature.

Reply:
I have updated this process. Hopefully, it suffices.